instrumentation/jaxrs/jaxrs-1.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JerseyTest.java

```diff
@@ -0,0 +1,127 @@
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class JerseyTest extends AbstractHttpServerUsingTest<Server> {
+
+  @RegisterExtension
+  public static final InstrumentationExtension testing =
+      HttpServerInstrumentationExtension.forAgent();
+
+  @BeforeAll
+  protected void setUp() {
+    startServer();
+  }
+
+  @AfterAll
+  protected void cleanUp() {
+    cleanupServer();
+  }
+
+  @Override
+  protected Server setupServer() throws Exception {
+    ServletContainer servlet =
+        new ServletContainer(
+            new Application() {
+              @Override
+              public Set<Object> getSingletons() {
+                Set<Object> objects = new HashSet<>();
+                objects.add(new Resource.Test1());
+                objects.add(new Resource.Test2());
+                objects.add(new Resource.Test3());
+                return objects;
+              }
+            });
+
+    ServletContextHandler handler = new ServletContextHandler(ServletContextHandler.SESSIONS);
+    handler.setContextPath(getContextPath());
+    handler.addServlet(new ServletHolder(servlet), "/*");
+
+    Server server = new Server(port);
+    server.setHandler(handler);
+    server.start();
+
+    return server;
+  }
+
+  @Override
+  protected void stopServer(Server server) throws Exception {
+    server.stop();
+    server.destroy();
+  }
+
+  @Override
+  protected String getContextPath() {
+    return "/jetty-context";
+  }
+
+  private static Stream<Arguments> provideArguments() {
+    return Stream.of(
+        Arguments.of("test/hello/bob", "/test/hello/{name}", "Test1", "hello", "Test1 bob!"),
+        Arguments.of("test2/hello/bob", "/test2/hello/{name}", "Test2", "hello", "Test2 bob!"),
+        Arguments.of("test3/hi/bob", "/test3/hi/{name}", "Test3", "hello", "Test3 bob!"),
+        Arguments.of("test3/nested", "/test3/nested", "Test3", "nested", "Test3 nested!"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideArguments")
+  void request(
+      String resource,
+      String expectedRoute,
+      String className,
+      String methodName,
+      String expectedResponse) {
+    AggregatedHttpResponse response =
+        client.post(address.resolve(resource).toString(), "bob").aggregate().join();
+    assertThat(response.status().code()).isEqualTo(200);
+    assertThat(response.contentUtf8()).isEqualTo(expectedResponse);
+
+    String controllerName = className + "." + methodName;
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("POST " + getContextPath() + expectedRoute)
```

Comment:
not related to your test, but it's strange to me that the groovy test used the [post method](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/71838a63d243d950156c689019e6621c227d45e9/instrumentation/jaxrs/jaxrs-1.0/javaagent/src/test/groovy/JerseyTest.groovy#L33) but asserted that the spans [were GETs](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/71838a63d243d950156c689019e6621c227d45e9/instrumentation/jaxrs/jaxrs-1.0/javaagent/src/test/groovy/JerseyTest.groovy#L42)

Reply:
In groovy test the server span isn't created from the actual request but with `runWithHttpServerSpan` that just uses `GET`.